Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp

```diff
@@ -306,41 +306,37 @@ namespace AzToolsFramework
         {
             // Only show the close icon if the prefab is expanded.
             // This allows the prefab container to be opened if it was collapsed during propagation.
-            if (!isExpanded)
+            if (isExpanded)
             {
-                return;
-            }
-
-            // Use the same color as the background.
-            QColor backgroundColor = m_backgroundColor;
-            if (isSelected)
-            {
-                backgroundColor = m_backgroundSelectedColor;
-            }
-            else if (isHovered)
-            {
-                backgroundColor = m_backgroundHoverColor;
-            }
+                // Use the same color as the background.
+                QColor backgroundColor = m_backgroundColor;
+                if (isSelected)
+                {
+                    backgroundColor = m_backgroundSelectedColor;
+                }
+                else if (isHovered)
+                {
+                    backgroundColor = m_backgroundHoverColor;
+                }
 
-            // Paint a rect to cover up the expander.
-            QRect rect = QRect(0, 0, 16, 16);
-            rect.translate(option.rect.topLeft() + offset);
-            painter->fillRect(rect, backgroundColor);
+                // Paint a rect to cover up the expander.
+                QRect rect = QRect(0, 0, 16, 16);
+                rect.translate(option.rect.topLeft() + offset);
+                painter->fillRect(rect, backgroundColor);
 
-            // Paint the icon.
-            QIcon closeIcon = QIcon(m_prefabEditCloseIconPath);
-            painter->drawPixmap(option.rect.topLeft() + offset, closeIcon.pixmap(iconSize));
+                // Paint the icon.
+                QIcon closeIcon = QIcon(m_prefabEditCloseIconPath);
+                painter->drawPixmap(option.rect.topLeft() + offset, closeIcon.pixmap(iconSize));
+            }
         }
         else
         {
             // Only show the edit icon on hover.
-            if (!isHovered)
+            if (isHovered)
```

Comment:
nit: combine else...if to else if

Reply:
I have another change that will merge cleanly by leaving it this way, so I will wait. But I'll keep in mind to clean this up :)